components/scream/src/interface/scream_scorpio_interface.F90

```diff
@@ -0,0 +1,955 @@
+  end type hist_var_list
+!----------------------------------------------------------------------
+  type pio_file_list
+    type(pio_atm_output), pointer :: pio_file => NULL()
+    type(pio_file_list),  pointer :: next => NULL()
+  end type pio_file_list
+  ! Define the first pio_file_list
+  type(pio_file_list), target :: pio_file_list_top
+!----------------------------------------------------------------------
+  type, public :: pio_atm_output
+        !> @brief Output filename.
+        character(len=max_chars) :: filename = ""
+
+        !> @brief Contains data identifying the file.
+        type(file_desc_t)     :: pioFileDesc
+
+        !> @brief Number of output dimensions, and counter to track them during
+        !  registration
+        integer               :: DimCounter = 0
+        !> @brief Recursive list of variables
+        type(hist_coord_list)   :: coord_list_top
+
+        !> @brief Number of output variable and counter to track them during
+        !  registration
+        integer               :: VarCounter = 0
+        !> @brief Recursive list of variables
+        type(hist_var_list)   :: var_list_top
+
+        !> @brief Number of history records on this file
+        integer               :: numRecs
+
+        !> @brief Coordinate Dimensions Array
+        type(hist_coord_t), allocatable :: dimensions(:)
+
+  end type pio_atm_output
+
+!----------------------------------------------------------------------
+  interface grid_write_data_array
+    module procedure grid_write_darray_1d_int
+    module procedure grid_write_darray_2d_int
+    module procedure grid_write_darray_3d_int
+    module procedure grid_write_darray_1d_double
+    module procedure grid_write_darray_2d_double
+    module procedure grid_write_darray_3d_double
+    module procedure grid_write_darray_1d_real
+    module procedure grid_write_darray_2d_real
+    module procedure grid_write_darray_3d_real
+  end interface
+!----------------------------------------------------------------------
+contains
+!=====================================================================!
+  subroutine eam_init_pio_1(mpicom,atm_id)
+
+    integer, intent(in) :: mpicom  ! MPI communication group for ATM
+    integer, intent(in) :: atm_id  ! Unique identifier assigned by component coupler
+
+    type(pio_atm_output), pointer :: current_atm_file => null()
+    integer :: ierr
+    logical :: found
+
+    ! Gather the pio_subsystem information for the atmosphere component.
+    call eam_init_pio_subsystem(mpicom,atm_id)
+
+    ! Allocate the dimension and variable arrays for the output file.
+    call get_pio_atm_file("example_pio_structured.nc",current_atm_file,found)
+    if (found) call errorHandle("PIO Error: Creating file: "//trim("example_pio_structured.nc")//" already exists",999)
+    current_atm_file%numRecs = 0
+
+    ! Set the filename TODO: Allow for mutliple files
+    current_atm_file%filename = "example_pio_structured.nc"
```

Comment:
weird that get_pio_atm_file doesn't set the file name already since you passed that file name. Maybe create a local variable for the file name, then pass that local variable to get_pio_atm_file and current_atm_file%filename so you don't have to define it twice? This is, of course, a super trivial comment.

Reply:
Yes, but it does make the code cleaner to move the assignment of the filename into `get_pio_atm_file`.  That's what I've done now